### ISSUE TITLE
[Data Table]: accessibility issues related to the component

### ISSUE THREAD

**Issue by mari-shakhanava** [epam/UUI#2320]
### Description
Open columns configuration modal button has no accessible name.

### Steps to Reproduce

1. Open Data Table component with 'showColumnsConfig={true}', 'allowColumnsReordering={true}' props or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Use T key to navigate to table.
3. Use the Tab key to navigate to [Open columns configuration modal] button and note the announcement.

### Actual result
![2024-06-05_16h08_27](https://github.com/epam/UUI/assets/70166849/c0428343-3739-450d-a1e3-b96209e268dd)

### Expected result
Open columns configuration modal button should be announced with accessible name that clearly describes its purpose.

Failed WCAG Checkpoint: 4.1.2 Name, Role, Value (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

### Description
Keyboard navigation via Tab key does not work for interactive columns.
Interactive columns in filtered table are not accessible via keyboard (it is impossible to navigate to columns' names via "Tab" key).
Note: Alternative navigation via shortcut "t" + arrows works as expected.

### Steps to Reproduce

1. Open Data Table component with filtersConfig passed as prop or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Use the Tab key to navigate to [Open columns configuration modal] button and note the announcement.

### Actual result
https://github.com/epam/UUI/assets/70166849/dbcd343d-1ce6-4899-8b7f-d86f8eea711c

### Expected result
Table's header should be accessible as well, because the majority of columns are interactive (there should be an opportunity to turn on sorting, to open filter dropdown without using a mouse).

Example of the correct behavior (navigation to sortable columns' names and visible focus):
https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/

### Description
When User navigates via keyboard inside dropdown menus, there is no opportunity to navigate outside it via keyboard (focus traps within dropdown list).

### Steps to Reproduce

1. Open Data Table component with filtersConfig passed as prop or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open filter dropdown.
3. Try to close it via "Esc" button.

### Actual result

https://github.com/epam/UUI/assets/70166849/9d9496af-4ba4-4105-9782-a59ce07cfef5

### Expected result
Pressing Esc key on any element of displayed listbox should close the listbox and move focus to the combobox.

Failed WCAG Checkpoint: 2.1.1 Keyboard (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

### Description
When User selects sorting direction in dropdown menu for any column, then there is no information about it for screen reader.

### Steps to Reproduce

1. Open Data Table component with filtersConfig passed as prop or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open filter dropdown.
3. Select either "Sort ascending" or "Sort descending" item and note the announcement.

### Actual result
![table-sort](https://github.com/epam/UUI/assets/70166849/5a918a29-6cd3-4aff-9b36-41f6e8922d89)
There is no difference for screen reader either sorting item is selected or not.

### Expected result
It should be clear for screen reader if any sorting direction is selected.

### Description
The Data Table does not have accessible name.

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Use T key to navigate to table and note the announcement.
3. Use the Tab key to navigate to [Open columns configuration modal] button and note the announcement.

### Actual result
![2024-06-24_12h29_01](https://github.com/epam/UUI/assets/70166849/b666d225-3eca-4b14-8de2-3a673ee601b8)
![table-name](https://github.com/epam/UUI/assets/70166849/868dc584-3a07-4208-a3ab-d1a090263b30)

The table does not have accessible name. Aria-label is missed.

### Expected result
The table has accessible name.

Example of the correct behavior:
https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/table/

[Success Criterion 4.1.2 Name, Role, Value (level A).](https://www.w3.org/WAI/WCAG22/quickref/?showtechniques=211#name-role-value)

### Description
When User opens any filter dropdown modal, then focus is on "Search" field instead of "Sort ascending" item.

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open any dropdown list.
3. Observe the keyboard focus position.

### Actual result

![Image](https://github.com/user-attachments/assets/74b0898a-a4ba-41ac-80f7-fc7bda0bb0d3)

Failed SC:
[2.4.3 Focus Order (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/focus-order)

### Expected result
"Sort Ascending" button should receive the focus as the first interactive element in the modal dialog.

### Description
User can't interact with checkboxes via keyboard

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open any dropdown list.
3. Try to navigate to the items in the dropdown list and check/unckeck any item using keyboard

### Actual result

https://github.com/user-attachments/assets/765f94a2-634b-4656-a6b1-593c8615da44
It is impossible to navigate to the dropdown list and check/uncheck any item using the keyboard.

Failed SC:
[2.1.1 Keyboard (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/keyboard)

### Expected result
It is possible to navigate to the dropdown list and check/uncheck any item using the keyboard. According to the requirements, checkboxes should be operated with "Tab" key (to navigate to the next checkbox) and "Space" key (to tick/untick the checkbox).

### Description
Tooltip text "Column header name" (on columns' names) disappears when the pointer moved over it.

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Hover mouse pointer over column's header -> tooltip text (column header name) is displayed.
3. Hover mouse pointer over tooltip text and observe.

### Actual result

Tooltip text with column header name  disappears when the pointer moved over it.

### Expected result
The pointer can be moved over the tooltip text without the tooltip text disappearing.
Failed WCAG Checkpoint: 1.4.13 Content on Hover or Focus (Level AA)
https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html

### Description
'Sort ascending/descending' buttons do not receive focus within filter dropdown menus (UUI version 6.1.4)

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open any dropdown list.
3. Try to navigate to the 'Sort ascending/descending' buttons using keyboard

### Actual result

Both buttons are not accessible via keyboard.

### Expected result
Buttons should be accessible via keyboard.
[Success Criterion 2.1.1 Keyboard](https://www.w3.org/WAI/WCAG22/Understanding/keyboard)

### Description
Chevron icon in the Table does not receive focus, which makes it impossible to expand a row using keyboard navigation.

### Steps to Reproduce
1. Open [Project Planning](https://uui.epam.com/demo?id=editableTable&theme=electric) demo.
2. Navigate through the rows using `Tab`.
3. Try to focus on the chevron icon.

### Actual result
Focus skips the сhevron icon; it cannot be reached with keyboard navigation.

https://github.com/user-attachments/assets/db1ad815-1b72-490f-b091-79099b48816e

### Expected result
The сhevron icon should be focusable with `Tab`.

* UUI version:  6.1.4
* Browser name: Crome latest
* Operating System: Windows latest

**Comment by NatalliaAlieva** [comment]
Comments:
Case №2 - ARIA pattern focuses the whole header (label + sort). Our table focuses only the sort control, not the column name. This limitation will remain as is.

**Comment by NatalliaAlieva** [comment]
Released in 6.4.0 ver.
